core/objects_test.go

```diff
@@ -144,3 +144,15 @@ func TestAuthorizationSolvedBy(t *testing.T) {
 		})
 	}
 }
+
+func TestFindChallengeByType(t *testing.T) {
```

Comment:
Could you also add a unit test for `Challenge.StringID` producing the correct hashes?